app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -86,42 +84,110 @@ async function addEthereumChainHandler(
     );
   }
 
-  let networkClientId;
-  let requestData;
-  let approvalFlowId;
-
-  if (!existingNetwork || existingNetwork.rpcUrl !== firstValidRPCUrl) {
-    ({ id: approvalFlowId } = await startApprovalFlow());
-
-    try {
-      await requestUserApproval({
-        origin,
-        type: ApprovalType.AddEthereumChain,
-        requestData: {
-          chainId,
-          rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
-          chainName,
-          rpcUrl: firstValidRPCUrl,
-          ticker,
-        },
-      });
+  const { id: approvalFlowId } = await startApprovalFlow();
+
+  let updatedNetwork;
+
+  try {
+    await requestUserApproval({
+      origin,
+      type: ApprovalType.AddEthereumChain,
+      requestData: {
+        chainId,
+        rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
+        chainName,
+        rpcUrl: firstValidRPCUrl,
+        ticker,
+      },
+    });
+
```

Comment:
It seems this new behavior will cause an approval to be shown to the user even when a existing network with the rpcUrl already exists where as the previous behavior does not. Is this new behavior intentional?

Reply:
Not intentional.  I'll try to to move the dupe check above startApprovalFlow.

Reply:
I made that change here: https://github.com/MetaMask/metamask-extension/pull/26433/commits/d440651d476088fba6c77913493664e58b1bc49b

If the rpc url and block explorer already exist, and are the defaults, an approval flow is no longer started.

If they exist but are not the defaults, I believe we want them to become the defaults.  But it seemed too sneaky to allow dapps to change that without a user prompt, so I've kept the approval flow in that case.